### PR TITLE
Fix sphinx links in sunpy/guide

### PR DESCRIPTION
**Pull request by dstansby**
Pulled out of https://github.com/sunpy/sunpy/pull/4178, should fix a lot of links in the guide docs.

### CODE REVIEW

File: docs/guide/acquiring_data/database.rst [hunk 1]
```diff
@@ -110,17 +110,17 @@ tags                   A list of :class:`tables.Tag` instances.
   ``observation_time_start``, ``observation_time_end``, ``instrument``,  ``size``,
   ``wavemin``, and ``wavemax`` are set by methods which use the VSO interface. In
   particular, these are :meth:`Database.add_from_vso_query_result`,
-  :meth:`Database.download` and   possibly :meth:`Database.fetch`.
+  ``Database.download`` and   possibly :meth:`Database.fetch`.
```
Comment (nabobalis) [inline docs/guide/acquiring_data/database.rst:113]: Is this not a method anymore?

**Comment by nabobalis** [comment]
Just a changelog and one question.

File: docs/guide/acquiring_data/database.rst (hunk 1 above)
Comment (dstansby) [reply docs/guide/acquiring_data/database.rst:113]: Doesn't look like it, possibly it was never a method?

Comment (nabobalis) [reply docs/guide/acquiring_data/database.rst:113]: Probably best to remove it then.

**Pull request merged**